sources/main.cpp

```diff
@@ -209,23 +209,36 @@ int main(int argc, char* argv[]) {
 
     // reset combination was pressed on startup
     if (buttonHandler->resetButtonsPressed()) {
-        // create marker file
-        QFile file("/firstrun");
-        if (file.open(QIODevice::WriteOnly)) {
-            file.close();
-        }
-
-        // restore default config
-        if (QFile::exists("/boot/config.json")) {
-            QFile::remove("/boot/config.json");
+        if (QFile::exists(qEnvironmentVariable(Environment::ENV_YIO_APP_DIR, appPath) + "/config.json") &&
+            QFile::exists("/boot/config.json")) {
```

Comment:
Attention: `/boot/config.json` is not required to exist. If it's missing the config.json from the app directory is used.
I.e. the check is not required, QFile::rename further down will silently fail and continue.